auth/authenticate-token.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require('jsonwebtoken');
+const secrets = require('../config/secrets.js');
+
+module.exports = {
+  genToken,
+  validateToken
+};
+
+function genToken(user) {
+  const payload = {
+    subject: user.id,
+    username: user.username
+  };
+  const secret = secrets.jwtSecret;
+  const options = {
+    expiresIn: '1hr'
+  };
+  return jwt.sign(payload, secret, options);
+}
+
+function validateToken(req, res, next) {
```

Comment:
Is this the same as authenticate middleware ?

Reply:
I separated the middleware to a different file. Token validation is now passing.